src/xlsx/mod.rs

```diff
@@ -2199,6 +2280,685 @@ pub(crate) fn path_to_zip_path<RS: Read + Seek>(zip: &ZipArchive<RS>, path: &str
+            key: QName(b"v"),
+            value,
+        } = a?
+        {
+            return Ok(Some(Box::from(value)));
+        }
+    }
+    Ok(None)
+}
+
+// Get the target location of the pivot table's pivot cache definitions.
+fn find_pivot_cache_definitions_from_pivot<RS>(
+    zip: &mut zip::ZipArchive<RS>,
+    path: &str,
+) -> Result<String, XlsxError>
+where
+    RS: Read + Seek,
+{
+    let (base_folder, file_name) = path.rsplit_once('/').expect("should be in a folder");
+    let rel_path = format!("{base_folder}/_rels/{file_name}.rels");
+    let mut xml = match xml_reader(zip, &rel_path) {
+        None => return Err(XlsxError::FileNotFound(rel_path.to_owned())),
+        Some(x) => x?,
+    };
+    let mut paths = vec![];
+    let mut buf = Vec::with_capacity(64);
+    loop {
+        buf.clear();
+        match xml.read_event_into(&mut buf) {
+            Ok(Event::Start(e)) if e.local_name().as_ref() == b"Relationship" => {
+                let mut target = String::new();
+                let mut is_pivot_cache_definitions_type = false;
+                for a in e.attributes() {
+                    match a? {
+                            Attribute {
+                                key: QName(b"Target"),
+                                value: v,
+                            } => target = xml.decoder().decode(&v)?.into_owned(),
+                            Attribute {
+                                key: QName(b"Type"),
+                                value: v,
+                            } => is_pivot_cache_definitions_type = *v == b"http://schemas.openxmlformats.org/officeDocument/2006/relationships/pivotCacheDefinition"[..],
+                            _ => (),
+                        }
+                }
+                if is_pivot_cache_definitions_type {
+                    if let Some(target) = target.strip_prefix("../") {
+                        // this is an incomplete implementation, but should be good enough for excel
+                        let (parent, _) = base_folder
+                            .rsplit_once('/')
+                            .expect("Must be a parent folder");
+                        paths.push(format!("{parent}/{target}"));
+                    } else if !target.is_empty() {
+                        paths.push(target);
+                    }
+                }
+            }
+            Ok(Event::End(e)) if e.local_name().as_ref() == b"Relationships" => break,
+            Ok(Event::Eof) => return Err(XlsxError::XmlEof("Relationships")),
+            Err(e) => return Err(XlsxError::Xml(e)),
+            _ => (),
+        }
+    }
+    if paths.len() > 1 {
+        Err(XlsxError::Unexpected(
+            "many definition cache relationships found for one pivot table",
+        ))
+    } else if paths.is_empty() {
+        Err(XlsxError::Unexpected(
+            "no cache definition found for pivot table",
+        ))
+    } else {
+        Ok(paths[0].clone())
+    }
```

Comment:
We should be able to apply the same solution to the other similar code path we rewrote, but I was fond of the stronger validation where we check there is only a single path that could match: I think using an `Option` is the best course of action here, if we find a duplicate we error.